tfjs-layers/src/engine/topology.ts

```diff
@@ -1018,7 +1018,31 @@ export abstract class Layer extends serialization.Serializable {
       // Actually call the layer, collecting output(s), mask(s), and shape(s).
       if (noneAreSymbolic) {
         let output = this.call(inputs as Tensor | Tensor[], kwargs);
-        // TODO(michaelterry): Compute the outputMask
+
+        // Apply masks to the output tensors if the layer supports it.
+        if (this.supportsMasking) {
+          // TODO(mattsoulanille): pass the input tensors' masks to computeMask
+          const outputMask = this.computeMask(inputs as Tensor | Tensor[]);
+          if (output instanceof Array && outputMask instanceof Array) {
+            if (output.length !== outputMask.length) {
+              throw new Error(`${this.name} output ${output.length} tensors `
+                + `but ${outputMask.length} masks for those tensors`);
+            }
+            for (let i = 0; i < output.length; i++) {
+              output[i].kerasMask = outputMask[i];
+            }
+          } else if (outputMask instanceof Array) {
```

Comment:
what if the array only contains one mask?

Reply:
I think that should be an error. If there's only one mask for all the tensors, it should be returned as a `tensor` instead of a `[tensor]`.

Reply:
Actually, keras does not seem to broadcast masks at all. Each output tensor needs its own mask:
https://github.com/keras-team/keras/blob/master/keras/engine/base_layer.py#L2893-L2898